### PR TITLE
add sharepoint details command

### PR DESCRIPTION
**Pull request by ryanfkeepers**
## Description

Adds the `backup details sharepoint` command
to the CLI.  E2E support is already functional.
Additional filters will be added as needs are
identified.

## Type of change

- [x] :sunflower: Feature

## Issue(s)

* #1614 

## Test Plan

- [x] :zap: Unit test

### CODE REVIEW

File: src/cli/utils/sharepoint.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
+package utils
+
+import (
+	"errors"
+
+	"github.com/alcionai/corso/src/pkg/selectors"
+)
+
+const (
+	LibraryItemFN = "library-item"
```
Comment (ryanfkeepers) [inline src/cli/utils/sharepoint.go:10]: @gmatev not sure what term we want to use for items here.  `library-item` is accurate, but a mouthful.  Granted, in the case of sharepoint, I expect users will much more rarely interact with restores on a per-item basis, so perhaps that's okay?

File: src/cli/backup/sharepoint.go [hunk 2]
```diff
@@ -282,3 +314,94 @@ func deleteSharePointCmd(cmd *cobra.Command, args []string) error {
...
+	if utils.HasNoFlagsAndShownHelp(cmd) {
+		return nil
+	}
+
+	s, acct, err := config.GetStorageAndAccount(ctx, true, nil)
+	if err != nil {
+		return Only(ctx, err)
+	}
+
+	r, err := repository.Connect(ctx, acct, s, options.Control())
```
Comment (dadams39) [inline src/cli/backup/sharepoint.go:346]: Nit. Runes can be quite hard to grok when looking through a large file like this.

File: src/cli/backup/sharepoint.go [hunk 3]
```diff
@@ -282,3 +314,94 @@ func deleteSharePointCmd(cmd *cobra.Command, args []string) error {
...
+		return nil
+	}
+
+	ds.PrintEntries(ctx)
+
+	return nil
+}
+
+// runDetailsSharePointCmd actually performs the lookup in backup details. Assumes
+// len(backupID) > 0.
```
Comment (dadams39) [inline src/cli/backup/sharepoint.go:377]: What is the consequence of this assumption?

File: src/cli/utils/onedrive_test.go [hunk 4]
```diff
@@ -4,12 +4,21 @@ import (
...
+	"github.com/stretchr/testify/suite"
 
 	"github.com/alcionai/corso/src/cli/utils"
 	"github.com/alcionai/corso/src/pkg/selectors"
 )
 
-func (suite *ExchangeUtilsSuite) TestIncludeOneDriveRestoreDataSelectors() {
+type OneDriveUtilsSuite struct {
+	suite.Suite
+}
```
Comment (dadams39) [inline src/cli/utils/onedrive_test.go:15]: What are we gaining by adding this suite?

File: src/cli/utils/sharepoint_test.go [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+			name: "no inputs",
+			opts: utils.SharePointOpts{
+				Sites:        empty,
+				LibraryPaths: empty,
+				LibraryItems: empty,
+			},
+			expectIncludeLen: 0,
+		},
+		{
+			name: "single inputs",
```
Comment (dadams39) [inline src/cli/utils/sharepoint_test.go:46]: typo `single input`

File: src/cli/utils/testdata/opts.go [hunk 6]
```diff
@@ -395,6 +395,94 @@ var (
...
+		// {
+		// 	Name: "EmptyFileCreatedBefore",
+		// 	Opts: utils.OneDriveOpts{
+		// 		FileCreatedBefore: "",
+		// 		Populated: utils.PopulatedFlags{
+		// 			utils.FileCreatedBeforeFN: struct{}{},
+		// 		},
+		// 	},
+		// },
+	}
```
Comment (dadams39) [inline src/cli/utils/testdata/opts.go:428]: Is there anything in this commented code that you wanted to be inspected or is this code scheduled to be removed?

File: src/pkg/backup/details/details.go [hunk 7]
```diff
@@ -352,12 +352,14 @@ func (i ExchangeInfo) Values() []string {
...
-	Created  time.Time `json:"created,omitempty"`
-	Modified time.Time `josn:"modified,omitempty"`
-	WebURL   string    `json:"webUrl,omitempty"`
-	Size     int64     `json:"size,omitempty"`
+	Created    time.Time `json:"created,omitempty"`
+	ItemName   string    `json:"itemName,omitempty"`
+	ItemType   ItemType  `json:"itemType,omitempty"`
+	Modified   time.Time `josn:"modified,omitempty"`
+	Owner      string    `json:"owner,omitempty"`
+	ParentPath string    `json:"parentPath"`
```
Comment (dadams39) [inline src/pkg/backup/details/details.go:360]: Not all sites have parents. omitempty?

**Review by dadams39 — APPROVED** [review]
Mostly nits, but nothing stopping the submission except for the comment surrounding `ParentPath`

File: src/cli/backup/sharepoint.go (hunk 2 above)
Comment (ryanfkeepers) [reply src/cli/backup/sharepoint.go:346]: Not sure what `runes` is referencing.

File: src/cli/backup/sharepoint.go (hunk 3 above)
Comment (ryanfkeepers) [reply src/cli/backup/sharepoint.go:377]: This comment is out of date.

File: src/cli/utils/onedrive_test.go (hunk 4 above)
Comment (ryanfkeepers) [reply src/cli/utils/onedrive_test.go:15]: Clarity about the set of tests being run, an alignment with standards elsewhere in the code where in all other cases (as far as I'm aware) suites are localized to a file, and a reduction in potential confusion about why onedrive tests are running as part of an exchange test suite.

File: src/pkg/backup/details/details.go (hunk 7 above)
Comment (ryanfkeepers) [reply src/pkg/backup/details/details.go:360]: Good call!

File: src/cli/backup/sharepoint.go (hunk 2 above)
Comment (dadams39) [reply src/cli/backup/sharepoint.go:346]: `runes` are single letters like `r` or `s`.

**Pull request merged**